### ISSUE TITLE
Method coverage result does not match with line coverage.

### ISSUE THREAD

**Issue by neurospeech** [axodox/AxoTools#56]
![capture](https://cloud.githubusercontent.com/assets/2217983/25565887/b9968c4c-2ded-11e7-8989-9bc958af0b39.PNG)

I have a method Signup and you can see on right hand side, my method's exception and one if condition isn't covered in unit test. In the Code Editor I can see correctly that my statements aren't covered in unit test. 

On left hand side tree, you can see that it displays 100% for the signup method, I think left side report is only considering whether the method is covered or not.

I have written unit test for only one method in entire project. There is no other class with Signup method.

**Comment by axodox** [comment]
I cannot reproduce this, I checked a large project with many exceptions, and seems to work correctly.

- Can you double click the method in the tree view, or press the source button when selected? This way we know it is the same method.
- Can you check if the merge-by-hash option is enabled in settings?
- Is the problem happening if you rerun the coverage report?

If that does not uncover the reason please try to make an example project demonstrating the issue.

**Comment by axodox** [comment]
Also if you check the image it says there is 3 lines in the method, while your method is clearly longer than that. Maybe it is still somehow a different method, e.g. an overload, or part of a base class.

Anyways double click it in the tree, and see where it leads you.

**Comment by neurospeech** [comment]
I found out the reason, method is same but it has an inline lambda. Most of code is inside inline lambda. Method is like...

```
       return Invoke( db => 
       {
                 ......     
                 ......
                 ......
       });

       public ActionResult Invoke(Func<AppDbContext,ActionResult> dbFunc){
             try{
                  ...
                  var db = .... getdb();
                  ...
                  return dbFunc(db);
             }
             catch(Exception ex){
                 ....
             }
       }
```

**Comment by axodox** [comment]
Lambda expressions result in new functions and classes so they are shown elsewhere.

**Comment by axodox** [comment]
So I think this is OK behavior - while these anonymous types and methods show up kind of ugly with their mangled names, but in other update I am grouping them which will organize things better. 

If you still have a problem please tell, otherwise I suggest to close the issue.

**Comment by neurospeech** [comment]
@axodox will update display correct coverage after adding up anonymous methods? I was able to drill down anonymous folder and was able to find fragments of method, but it is just difficult to calculate total coverage, you can close this for now and if in new update if things aren't clear then I will reopen the issue.

**Comment by axodox** [comment]
No, the coverage is correct right now and there is no plan to sum up these lambdas.

I will take your example to explain you why:
- we see that Invoke is a normal function outside your method
- we also see that db => ... is a function, which is defined in your method, but it is not part of it - e.g. you can store it, and use it elsewhere, just like other methods. 

While you could say: but I can use my local variables inside db => ...
That is however just syntactic sugar, as in reality the compiler adds a class and stores the data and your db => ... method there.

Because of this we cannot sum db => ... coverage up to your method. As it could be called from multiple methods which would result in higher coverage than it really is. Just remember that in reality there is no difference in how Invoke and db => ... is called internally.

That being said I agree that the problem you describe is real - it is harder to read the coverage like this. But it is also harder to read the stack trace. I have a simple solution: limit your lambdas to short length, otherwise just define an ordinary method in your class, or if you need to store data, define a class to store it and use one of its methods to do the job. This is actually what the compiler does for you!

**Comment by neurospeech** [comment]
I believe this is still solvable as method has line number information from where it starts and where it ends, code editor perfectly displays lines that are covered and lines that are not, if you try to sum up lines, you might be able to get correct result.

**Comment by axodox** [comment]
So what you want is that if a anonymous method / lambda expression is defined inside another method, we should remove it from results, and add its branch and coverage points to the method defining it?

This would obviously have nothing to do with its being called from the given method or not. E.g. the method could put it in a queue, from which another thread could execute it later. Also lambdas outside methods would remain there as anonymous methods.

It would work though... Ok I might consider adding this as an option, but I would take it as low prio for now.

**Comment by axodox** [comment]
If my understanding is right we can add a new enhancement issue for this. We could title it: "Add option to include lambda coverage in parent method for easier comprehension of coverage report" - well something like this but shorter...

**Comment by neurospeech** [comment]
Yes, Methods today contain lambda expressions and think of new inline functions included in c# 7.0, this will be important for inline functions as well. I think added option would be nice.
